hasher-matcher-actioner/webapp/src/utils/TextFieldsUtils.jsx

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+ */
+
+import React, {useState, useEffect, useCallback} from 'react';
+import {Popover, OverlayTrigger} from 'react-bootstrap';
+import copy from 'clipboard-copy';
+
+function truncateString(input, maxLength = 25) {
+  if (input == null) {
+    return '';
+  }
+  if (input.length <= maxLength) {
+    return input;
+  }
+  return `${input.substring(0, maxLength)}...`;
+}
+
+// eslint-disable-next-line react/prop-types
+export function CopyableHashField({text, tooltip}) {
+  return (
+    <td style={{maxWidth: '250px', overflow: 'hidden'}}>
+      <span style={{overflow: 'hidden'}} />
+      <CopyableTextField
+        text={text}
+        tooltip={tooltip ?? 'Copy hash to clipboard?'}
+      />
+    </td>
+  );
+}
+
+const UpdatingPopover = React.forwardRef(
```

Comment:
Whats a popover? Is this in the demo video?

Reply:
popover is what react-boostrap uses for dynamic content tool tips